src/main/java/httpserver/router/Router.java

```diff
@@ -20,20 +20,24 @@ public Router() {
         routes.put("/simple_get", new String[]{Methods.GET, Methods.HEAD});
         routes.put("/simple_get_with_body", new String[]{Methods.GET});
         routes.put("/head_request", new String[]{Methods.HEAD});
+        routes.put("/method_options", new String[]{Methods.GET, Methods.HEAD, Methods.OPTIONS});
+        routes.put("/method_options2", new String[]{Methods.GET, Methods.HEAD, Methods.OPTIONS, Methods.POST, Methods.PUT});
         return routes;
     }
 
     public boolean isRequestValid(Request request) {
-        for (Map.Entry<String, String[]> set : routes.entrySet()) {
-            if (set.getKey().equals(request.path)) {
-                String[] methodsForPath = set.getValue();
-                for (int i = 0; i < methodsForPath.length; i++) {
-                    if (methodsForPath[i].equals(request.method)) {
-                        return true;
-                    }
+        if (routes.containsKey(request.path)) {
+            String[] methodsForPath = routes.get(request.path);
+            for (int i = 0; i < methodsForPath.length; i++) {
+                if (methodsForPath[i].equals(request.method)) {
+                    return true;
                 }
```

Comment:
Glad you were able to get rid of the nested for loop!

Reply:
This is neat syntax to try out sometime: 
for (int element : arr) {
    if (element == toCheckValue) {
        return true;
    }
}